dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdMappingFileUploadTask.kt

```diff
@@ -179,5 +193,61 @@ open class DdMappingFileUploadTask
         return true
     }
 
+    private fun shrinkMappingFile(
+        mappingFile: File
+    ): File {
+        if (!mappingFileTrimIndents && mappingFilePackagesAliases.isEmpty()) return mappingFile
+
+        val shrinkedFile = File(mappingFile.parent, "mapping-shrinked.txt")
+        // sort is needed to have predictable replacement in the following case:
+        // imagine there are 2 keys - "androidx.work" and "androidx.work.Job", and the latter
+        // occurs much often than the rest under "androidx.work.*". So for the more efficient
+        // compression we need first to process the replacement of "androidx.work.Job" and only
+        // after that any possible prefix (which has a smaller length).
+        val replacements = mappingFilePackagesAliases.entries
+            .sortedByDescending { it.key.length }
+            .map {
+                Regex("${it.key}(?=\\W)") to it.value
+            }
+
+        mappingFile.readLines()
+            .map {
+                if (mappingFileTrimIndents) it.trimStart() else it
+            }
+            .map {
+                applyShortAliases(it, replacements)
+            }
+            .forEach {
+                shrinkedFile.appendText(it + "\n")
+            }
+
+        return shrinkedFile
+    }
+
+    private fun applyShortAliases(
+        line: String,
+        replacements: List<Pair<Regex, String>>
+    ): String {
+        val isLineWithRename = !line.startsWith(MAPPING_FILE_COMMENT_CHAR) &&
```

Comment:
so we can dont rename `synthetized` comment even if we are renaming the fields on which the comment apply ?

Reply:
`Synthesized` comment doesn't carry the name, it is just a directive placed below the frame saying that it should be removed from retraced stack according to https://r8.googlesource.com/r8/+/refs/heads/main/doc/retrace.md#synthesized-introduced-at-version-1_0.

It seems only `RewriteFrame` comment can carry the name (in the `conditions` field), but I didn't find such comments in the mapping file of Datadog app, so I guess they are pretty rare, and even if we renamed something which stayed non-renamed in the `conditions`, worst case scenario will be just that `RewriteFrame` directive won't be applied.